frontend/public/components/terminal.jsx

```diff
@@ -3,15 +3,19 @@ import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import { Terminal as XTerminal } from 'xterm';
 import * as fit from 'xterm/lib/addons/fit/fit';
+import * as full from 'xterm/lib/addons/fullscreen/fullscreen';
 
 XTerminal.applyAddon(fit);
+XTerminal.applyAddon(full);
 
 export class Terminal extends React.Component {
   constructor(props) {
     super(props);
     this.state = {height: 0, width: 0};
     this.innerRef = React.createRef();
     this.outerRef = React.createRef();
+    this.isFullscreen = false;
+    this.fullscreenColOffset = -40;
```

Comment:
I'm confused by this magic number. Why is it needed? (I know we have the extra margin, but why? And why 40px? Will this value always be correct?)

I'd rather punt on this until we understand it better.

Reply:
I knew you'd comment on this :wink:  Yes, it is some fuzzy logic.  It's not a margin per say, rather the #Cols the xterm terminal derives from the width.  I'll remove/punt on this until we understand what the xterm addon is doing.